core/server/lib/image/cached-image-size-from-url.js

```diff
@@ -1,49 +1,54 @@
 const debug = require('ghost-ignition').debug('utils:image-size-cache');
-const imageSize = require('./image-size');
 const errors = require('@tryghost/errors');
-const logging = require('../../../shared/logging');
-const cache = {};
 
-/**
- * Get cached image size from URL
- * Always returns {object} imageSizeCache
- * @param {string} url
- * @returns {Promise<Object>} imageSizeCache
- * @description Takes a url and returns image width and height from cache if available.
- * If not in cache, `getImageSizeFromUrl` is called and returns the dimensions in a Promise.
- */
-function getCachedImageSizeFromUrl(url) {
-    if (!url || url === undefined || url === null) {
-        return;
+class CachedImageSizeFromUrl {
+    constructor({logging, imageSize}) {
+        this.logging = logging;
+        this.imageSize = imageSize;
+        this.cache = new Map();
     }
 
-    // image size is not in cache
-    if (!cache[url]) {
-        return imageSize.getImageSizeFromUrl(url).then(function (res) {
-            cache[url] = res;
-
-            debug('Cached image:', url);
-
-            return cache[url];
-        }).catch(errors.NotFoundError, function () {
-            debug('Cached image (not found):', url);
-            // in case of error we just attach the url
-            cache[url] = url;
-
-            return cache[url];
-        }).catch(function (err) {
-            debug('Cached image (error):', url);
-            logging.error(err);
-
-            // in case of error we just attach the url
-            cache[url] = url;
-
-            return cache[url];
-        });
+    /**
+     * Get cached image size from URL
+     * Always returns {object} imageSizeCache
+     * @param {string} url
+     * @returns {Promise<Object>} imageSizeCache
+     * @description Takes a url and returns image width and height from cache if available.
+     * If not in cache, `getImageSizeFromUrl` is called and returns the dimensions in a Promise.
+     */
+    getCachedImageSizeFromUrl(url) {
```

Comment:
Worth marking this method as `async` as well as long as return type is `Promise<Object>`

Reply:
`async` just enables to use `await` in the function, right? In that case I'll also refactor the function to use `async` + `await` in a follow up PR.